src/gui/qgisinterface.h

```diff
@@ -889,10 +889,22 @@ class GUI_EXPORT QgisInterface : public QObject
     /**
      * Adds a \a dock widget to the main window, in the specified dock \a area.
      *
+     * \see addTabifiedDockWidget()
      * \see removeDockWidget()
      */
     virtual void addDockWidget( Qt::DockWidgetArea area, QDockWidget *dockwidget ) = 0;
 
+    /**
+     * Adds a \a dock widget to the given \a area and tabify it (if other dock widgets
+     * exist in the same \a area). The new tab will be selected.
+     *
+     * If \a onTopOf is given, the \a dock widget will be placed on top of a dock
+     * widget with such name (if it exists).
+     *
+     * \see addDockWidget()
+     */
+    virtual void addTabifiedDockWidget( Qt::DockWidgetArea area, QDockWidget *dockwidget, const QString &onTopOf = QString() ) = 0;
```

Comment:
I wonder if onTopOf should be an list (ordered by priority). It would make this more adaptable to customised dock placement...

Reply:
Yes, I'm fine with a list of object names here. 
Implemented in https://github.com/qgis/QGIS/pull/35933/commits/14ba75517310d185c1195b7027d674e9ff4b5889!